src/plugins/intel_npu/tests/unit/CMakeLists.txt

```diff
@@ -29,6 +29,8 @@ ov_add_test_target(
             ${OpenVINO_SOURCE_DIR}/src/plugins/intel_npu/src/utils/include
             ${OpenVINO_SOURCE_DIR}/src/plugins/intel_npu/src/plugin/include
             ${OpenVINO_SOURCE_DIR}/src/plugins/intel_npu/src/al/include
+        OBJECT_FILES
+            ${OpenVINO_SOURCE_DIR}/src/plugins/intel_npu/src/plugin/src/metadata.cpp
```

Comment:
Not great with CMake but gotta ask: why did we need this? Asking cause it looks a bit out-of-the-blue to me.

Reply:
The source files from `intel_npu/src/plugin/src` are not compiled into a static library. That means I cannot add them to the unit tests, without doing some stuff in our root CMakeFiles.
Another solution would imply moving `metadata.cpp/hpp` to `common` dir which is already compiled into a static lib.

But, the quickest solution was to hardcode the needed source file for my PR.

Reply:
@pereanub Could you please check everything is fine here?

Reply:
Was informed in a meeting we're fine with this.